core/src/container.cpp

```diff
@@ -412,11 +412,13 @@ void SerialContainer::onNewFailure(const Stage& child, const InterfaceState* fro
 			// TODO: If child is a container, from and to might have associated solutions already!
 			break;
 
+		// TODO(v4hn): Only disable partial incoming solution if there are no other solutions possible
+		// (Propagators might generate multiple solutions per incoming state)
```

Comment:
I agree on this statement. But I would expect that a new feasible solution generated later will reactivate the whole solution branch again. Thus, we disable only temporarily. Doesn't that work?

Reply:
Yes and no.

The other way around is the crucial failure: a new failure might disable a partial solution even if there is already valid alternatives for the branch. So if the last propagated solution for an incoming state is a failure all others will also stay disabled for no reason. In practice, a successful extension in the next stage might reactive it, but we can't guarantee the next stage can compute before we compute the disabled branch. That would be hell of a bug to search for.

Reply:
I agree, but didn't you fix this in 2b759570a6edc48462b0930418c7999e6793fefd?

Reply:
You are right. There was one other issue, but I added a separate test (and fix) for that now.